yeat/workflows/snakefiles/Paired

```diff
@@ -158,12 +158,14 @@ rule megahit:
         read2="seq/downsample/{sample}/{sample}.R2.fq.gz"
     threads: 128
     params:
+        temp_dir=lambda wildcards: f"analysis/{wildcards.sample}/{wildcards.label}/megahit-temp",
+        actual_dir=lambda wildcards: f"analysis/{wildcards.sample}/{wildcards.label}/megahit",
         extra_args=lambda wildcards: config["extra_args"][wildcards.label]
     shell:
         """
-        megahit -1 {input.read1} -2 {input.read2} -t {threads} -o analysis/{wildcards.sample}/{wildcards.label}/megahit-temp {params.extra_args}
-        mv analysis/{wildcards.sample}/{wildcards.label}/megahit-temp/* analysis/{wildcards.sample}/{wildcards.label}/megahit
-        rm -r analysis/{wildcards.sample}/{wildcards.label}/megahit-temp
+        megahit -1 {input.read1} -2 {input.read2} -t {threads} -o {params.temp_dir} {params.extra_args}
+        mv {params.temp_dir}/* {params.actual_dir}
+        rm -r {params.temp_dir}
```

Comment:
Yes, I like the result from moving some of these complex parameters into the `params` block to clean up the shell syntax.